vita3k/codec/src/aac.cpp

```diff
@@ -63,96 +38,69 @@ uint32_t AacDecoderState::get(DecoderQuery query) {
 }
 
 bool AacDecoderState::send(const uint8_t *data, uint32_t size) {
-    packet_cache.emplace_back(data, data + size); // add to packet cache
-
-    // format context needs to be built when there is at least some data
-    if (!format_context) {
-        format_context = std::shared_ptr<AVFormatContext>(avformat_alloc_context(), &avformat_free_context);
-        AVFormatContext *format_ptr = format_context.get();
-        assert(format_ptr);
-
-        format_ptr->pb = io_context.get();
-
-        int err = avformat_open_input(&format_ptr, "", nullptr, nullptr);
-        assert(err == 0);
+    AVPacket *packet = av_packet_alloc();
+    packet->data = const_cast<uint8_t *>(data);
+    packet->size = size;
 
-        // AVCodec *codec = nullptr;
-        // stream_id = av_find_best_stream(format_ptr, AVMEDIA_TYPE_AUDIO, -1, -1, nullptr, 0);
-        // assert(stream_id >= 0);
-        // assert(context->codec == codec);
-    }
+    av_frame_unref(frame);
 
-    AVPacket *packet = av_packet_alloc();
-    int err = av_read_frame(format_context.get(), packet);
-    if (err != 0) {
-        LOG_INFO("Error reading frame: {}.", codec_error_name(err));
-        return false;
-    }
+    int got_frame;
+    int len = codec->decode(context, frame, &got_frame, packet);
+    assert(got_frame);
 
-    err = avcodec_send_packet(context, packet);
     av_packet_free(&packet);
-    if (err < 0) {
-        LOG_WARN("Error sending Aac packet: {}.", codec_error_name(err));
+    if (len < 0) {
+        LOG_WARN("Error sending Aac packet.");
         return false;
     }
 
+    es_size_used = static_cast<uint32_t>(len);
+
     return true;
 }
 
 bool AacDecoderState::receive(uint8_t *data, DecoderSize *size) {
-    AVFrame *frame = av_frame_alloc();
-
-    int err = avcodec_receive_frame(context, frame);
-    if (err < 0) {
-        LOG_WARN("Error receiving Aac frame: {}.", codec_error_name(err));
-        av_frame_free(&frame);
-        return false;
-    }
-
     assert(frame->format == AV_SAMPLE_FMT_FLTP);
 
     if (data) {
         convert_f32_to_s16(
             reinterpret_cast<float *>(frame->extended_data),
             reinterpret_cast<int16_t *>(data),
-            frame->channels, frame->nb_samples,
-            frame->sample_rate);
+            context->channels, frame->nb_samples,
+            context->sample_rate);
     }
 
     if (size) {
         size->samples = frame->nb_samples;
     }
 
-    av_frame_free(&frame);
     return true;
 }
 
+uint32_t AacDecoderState::get_es_size() {
+    return es_size_used;
+}
+
 AacDecoderState::AacDecoderState(uint32_t sample_rate, uint32_t channels) {
     constexpr size_t buffer_size = 8192;
```

Comment:
not used

Reply:
is used

Reply:
Used and necessary here https://github.com/Zangetsu38/Vita3K/blob/a5a45e967275fe976890c3680c8a691daef902d6/vita3k/modules/SceAudiodec/SceAudiodecUser.cpp#L207

Reply:
I comment out 
constexpr size_t buffer_size = 8192;
and got no errors

Reply:
You highlighted get_es_size, but otherwise I agree that buffer_size is useless.

Reply:
ah yes buffer_size is not used, you ping on get es size line hdre lol